pkg/util/pravegacluster.go

```diff
@@ -85,8 +86,42 @@ func ConfigMapNameForSegmentstore(clusterName string) string {
 	return fmt.Sprintf("%s-pravega-segmentstore", clusterName)
 }
 
-func StatefulSetNameForSegmentstore(clusterName string) string {
-	return fmt.Sprintf("%s-pravega-segmentstore", clusterName)
+//function to check if the version is below 0.7 or not
+func IsVersionBelow07(ver string) bool {
+	if ver == "" {
+		return true
+	}
+	result, _ := CompareVersions(ver, "0.7.0", "<")
+	if result {
+		return true
+	}
+	return false
+}
+
+//this function will return true only in case of upgrading from a version below 0.7 to pravega version 0.7 or later
+func IsClusterUpgradingTo07(p *api.PravegaCluster) bool {
+	if !IsVersionBelow07(p.Spec.Version) && IsVersionBelow07(p.Status.CurrentVersion) {
+		return true
+	}
+	return false
+}
+
+//if version is above or equals to 0.7 this name will be assigned
+func StatefulSetNameForSegmentstoreAbove07(name string) string {
+	return fmt.Sprintf("%s-pravega-segment-store", name)
```

Comment:
This change just broke the upgrade.

Reply:
`0.7.0` pravega clusters historically were deployed with `0.4.x` p-operator and had the stateful sets named using the `pravega-segmentstore` substring.